mypy/checker.py

```diff
@@ -1845,6 +1851,11 @@ def check_compatibility_all_supers(self, lvalue: RefExpr, lvalue_type: Optional[
 
                 base_type, base_node = self.lvalue_type_from_base(lvalue_node, base)
 
+                if isinstance(base_node, TypeAlias):
```

Comment:
Similar to above, should we check for other kinds of things that shouldn't be overridden?

Why do we this check in two places? If we need to have the check in two places, it may be slightly better to move it to a helper method instead of repeating the code, at least if the check becomes more complex.